crates/primitives/starknet/src/transaction/mod.rs

```diff
@@ -753,7 +756,7 @@ impl Transaction {
         AccountTransactionContext {
             transaction_hash: tx.transaction_hash(),
             max_fee: tx.max_fee(),
-            version: TransactionVersion(StarkFelt::from(1_u8)),
+            version: calculate_transaction_version_from_u8(self.is_query, 1_u8),
```

Comment:
`tx.version` no?

Reply:
Actually `InvokeTransaction` doesn't have a version field which is why this is hardcoded to 1. And v0 is deprecated and we don't handle it. However, in case of v0 our code will mostly fail with the wrong message (I guess invalid signature). So I spoke to greg and was thinking to create a new issue to throw a good error for v0.

Reply:
please do

Reply:
Ok nevermind, we do throw an error for V0 transactions [here](https://github.com/keep-starknet-strange/madara/blob/b99d8bc5a1aa4449bfbe8ddde0765dd802856f80/crates/primitives/starknet/src/transaction/types.rs#L624)